v-next/hardhat/src/internal/cli/init/init.ts

```diff
@@ -45,9 +193,195 @@ async function throwIfCwdAlreadyInsideProject() {
+  force?: boolean,
+): Promise<void> {
+  // Find all the files in the workspace that would have been overwritten by the template files
+  const matchingFiles = await getAllFilesMatching(workspace, (file) =>
+    template.files.includes(path.relative(workspace, file)),
+  ).then((files) => files.map((f) => path.relative(workspace, f)));
+
+  // Ask the user for permission to overwrite existing files if needed
+  if (matchingFiles.length !== 0) {
+    if (force === undefined) {
+      force = await promptForForce(matchingFiles);
+    }
+  }
+
+  // Copy the template files to the workspace
+  for (const file of template.files) {
+    if (force === false && matchingFiles.includes(file)) {
+      continue;
+    }
+    const pathToTemplateFile = path.join(template.path, file);
+    const pathToWorkspaceFile = path.join(workspace, file);
+
+    await ensureDir(path.dirname(pathToWorkspaceFile));
+    await copy(pathToTemplateFile, pathToWorkspaceFile);
+  }
+
+  console.log(`✨ ${chalk.cyan(`Template files copied`)} ✨`);
+}
+
+/**
+ * installProjectDependencies prints the commands to install the project dependencies
+ * and runs them if the install option is true or if the user opts-in to it.
+ *
+ * NOTE: This function is exported for testing purposes
+ *
+ * @param workspace The path to the workspace to initialize the project in.
+ * @param template The template to use for the project initialization.
+ * @param install Whether to install the project dependencies.
+ */
+export async function installProjectDependencies(
+  workspace: string,
+  template: Template,
+  install?: boolean,
+): Promise<void> {
+  const pathToWorkspacePackageJson = path.join(workspace, "package.json");
+
+  const workspacePkg: PackageJson = await readJsonFile(
+    pathToWorkspacePackageJson,
+  );
+
+  const hardhatVersion = await getHardhatVersion();
+  const packageManager = await getPackageManager(workspace);
+
+  // Find the template dev dependencies that are not already installed
+  const templateDependencies = template.packageJson.devDependencies ?? {};
+  // If the package manager doesn't install peer dependencies by default,
+  // we need to add them to the template dependencies
+  if (!(await installsPeerDependenciesByDefault(workspace, packageManager))) {
+    const templatePeerDependencies =
+      template.packageJson.peerDependencies ?? {};
+    for (const [name, version] of Object.entries(templatePeerDependencies)) {
+      templateDependencies[name] = version;
+    }
+  }
+  const workspaceDependencies = workspacePkg.devDependencies ?? {};
+  const dependenciesToInstall = Object.entries(templateDependencies)
+    .filter(([name]) => workspaceDependencies[name] === undefined)
+    .map(([name, version]) => {
+      // If the version is workspace:, replace it with the current version of Hardhat
+      if (version.startsWith("workspace:")) {
+        return `"${name}@${hardhatVersion}"`;
+      }
+      return `"${name}@${version}"`;
+    });
```

Comment:
This part won't work, as the versions of every package in the mono repo are independent. I think it's better not to use `workspace:` in the templates and handle those versions manually (i.e. bump them when appropriate).

Reply:
Thanks for documenting the behavior in the `README.md`, btw.

That would be useful in the future.

Reply:
Ay, of course, I forgot that we don't do mass-releases of all the packages. I removed replacing the `workspace:` prefixed versions with current hardhat version. Instead, I added stripping `workspace:` prefixes. That way we should still be able to quickly test the templates with the workspace packages while delivering concrete versions (that we have full control over) to our users. In practice, during project initialization:

```json
  "devDependencies": {
    "@ignored/hardhat-vnext": "workspace:^3.0.0-next.3",
    "typescript": "~5.5.0"
  }
```

would become

```json
  "devDependencies": {
    "@ignored/hardhat-vnext": "^3.0.0-next.3",
    "typescript": "~5.5.0"
  }
```

Let's iterate on it in a follow-up if needed.